test/functional/wallet_upgradewallet.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""upgradewallet RPC functional test
+
+Test upgradewallet RPC. Download v0.16.3 node binary:
+
+contrib/devtools/previous_release.sh -b v0.16.3
+"""
+
+import os
+import shutil
+
+from test_framework.test_framework import BitcoinTestFramework, SkipTest
+from test_framework.util import (
+    assert_equal,
+    assert_greater_than
+)
+
+class UpgradeWalletTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [
+            ["-addresstype=bech32"],
+            []
+        ]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def setup_nodes(self):
+        if os.getenv("TEST_PREVIOUS_RELEASES") == "false":
+            raise SkipTest("upgradewallet RPC tests")
+
+        releases_path = os.getenv("PREVIOUS_RELEASES_DIR") or os.getcwd() + "/releases"
+        if not os.path.isdir(releases_path):
+            if os.getenv("TEST_PREVIOUS_RELEASES") == "true":
+                raise AssertionError("TEST_PREVIOUS_RELEASES=1 but releases missing: " + releases_path)
+            raise SkipTest("This test requires binaries for previous releases")
+
+        self.add_nodes(self.num_nodes, extra_args=self.extra_args, versions=[
+            None,
+            160300
+        ], binary=[
+            self.options.bitcoind,
+            releases_path + "/v0.16.3/bin/bitcoind",
+        ], binary_cli=[
+            self.options.bitcoincli,
+            releases_path + "/v0.16.3/bin/bitcoin-cli",
+        ])
+
+        self.start_nodes()
```

Comment:
Instead of modifying the test_framwork/util.py, you could just remove the line `[regtest]` from the bitcoin.conf in the 16.3 datadir before starting the nodes

Reply:
Yes, but wouldn't be better to make this generally available. For example the [comment](https://github.com/bitcoin/bitcoin/blob/master/test/functional/feature_backwards_compatibility.py#L11) in feature_backwards_compatibility states that it's not possible to test older nodes.

Reply:
Yes, if you want, you can create a generally available util method like

```py
def adjust_bitcoin_conf_for_pre_16_3(conf_file):
  ...

Reply:
Ok, will implement it. Have just force pushed a change that uses *sed* to make an in-file replacement.